### PR TITLE
fix: Flush wavefront output sender on error to clean up broken connections

### PR DESCRIPTION
**Pull request by mikolajprzybysz**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves https://github.com/influxdata/telegraf/pull/8812

1. update wavefront-sdk-go to the most recent version
2. added flushing of the sender once it encounters an error, without it what's happening is that the connection is never cleaned and the client is constantly trying to send messages via broken connection. This means if the wavefront-proxy goes down and the connection breaks (for example when container has crushed or you are simply deploying new version of it) telegraf will never again acquire new connection and resume sending, with this fix it will

### CODE REVIEW

**Comment by mikolajprzybysz** [comment]
!signed-cla

**Comment by sfc-gh-mprzybysz** [comment]
!signed-cla

File: plugins/outputs/wavefront/wavefront.go [hunk 1]
```diff
@@ -173,7 +173,8 @@ func (w *Wavefront) Write(metrics []telegraf.Metric) error {
 			err := w.sender.SendMetric(point.Metric, point.Value, point.Timestamp, point.Source, point.Tags)
 			if err != nil {
 				if isRetryable(err) {
-					return fmt.Errorf("wavefront sending error: %v", err)
+					w.Log.Errorf("wavefront sending error: %v", err)
+					return w.sender.Flush()
```
Comment (srebhan) [inline plugins/outputs/wavefront/wavefront.go:177]: To be honest I would prefer
```suggestion
					if flushErr := w.sender.Flush(); flushErr != nil {
						w.Log.Errorf("wavefront flushing error: %v", flushErr)
					}
					return err
```
as the sending error is the relevant one to return IMO. What do you think?

Comment (srebhan) [inline plugins/outputs/wavefront/wavefront.go:177]: To be honest I would prefer
```suggestion
					if flushErr := w.sender.Flush(); flushErr != nil {
						w.Log.Errorf("wavefront flushing error: %v", flushErr)
					}
					return err
```
as the sending error is the relevant one to return IMO. What do you think?

**Review by srebhan — COMMENTED** [review]
Thanks for fixing this issue @mikolajprzybysz! One comment about which error to return. What do you think?

File: plugins/outputs/wavefront/wavefront.go (hunk 1 above)
Comment (sfc-gh-mprzybysz) [reply plugins/outputs/wavefront/wavefront.go:177]: Yes, you are right, if the sending error is omitted it's possible to lost the remaining metrics from the given metric batch.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @mikolajprzybysz for the quick fix!

**Review by powersj — APPROVED** [review]

**Pull request merged**